src/main/java/com/marklogic/kafka/connect/source/DslQueryHandler.java

```diff
@@ -0,0 +1,84 @@
+package com.marklogic.kafka.connect.source;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.datamovement.RowBatcher;
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.row.RawQueryDSLPlan;
+import com.marklogic.client.row.RowManager;
+import org.springframework.util.StringUtils;
+
+import java.util.Map;
+
+public class DslQueryHandler extends LoggingObject implements QueryHandler {
```

Comment:
I'll look at this more closely next week, but one thing I like right away is that there aren't any Kafka dependencies in here (the config map is related to Kafka, but that could easily be broken out into the actual params that the handler needs). I think that's important because this capability - modifying a user's query and then getting the highest value for a particular column - is something we may want to migrate into the Java Client itself. For example, the similar logic for doing that in the NiFi connector with the Search API could arguably in the Java Client too.